perfkitbenchmarker/disk.py

```diff
@@ -45,6 +93,67 @@ def __init__(cls, name, bases, dct):
     super(AutoRegisterDiskSpecMeta, cls).__init__(name, bases, dct)
 
 
+def WarnAndTranslateDiskTypes(name, cloud):
+  """Translate old disk types to new disk types, printing warnings if needed.
+
+  Args:
+    name: a string specifying a disk type, either new or old.
+    cloud: the cloud we're running on.
+
+  Returns:
+    The new-style disk type name (i.e. the provider's name for the type).
+
+  Raises:
+    ValueError, if name is not a disk type name.
+  """
+
+  if cloud in DISK_TYPE_MAPS:
+    disk_type_map = DISK_TYPE_MAPS[cloud]
+    if name in disk_type_map and disk_type_map[name] != name:
+      new_name = disk_type_map[name]
+      logging.warning('Disk type name %s is deprecated and will be removed. '
+                      'Translating to %s for now.', name, new_name)
+      return new_name
+    else:
+      return name
+  else:
```

Comment:
Yes, for providers that have not been updated for the disk types refactor. This lets us port providers gradually rather than all at once.